### ISSUE TITLE
Sprint Aug 10

### ISSUE THREAD

**Issue by diasdavid** [ipfs/pm#26]
## Sprint Discussions

See https://github.com/ipfs/pm#sprint-discussion-schedule

## Sprint Deliverables

- Add yourself below, that way we only have people listed who are really in the sprint
- Add links to issues down here. Only add things you can finish this sprint.

### @jbenet

- infrastructure
  - [ ] ssl cert for gateway.ipfs.io
- electron app
  - [ ] design dropdown menu
  - [ ] implement dropdown menu
  - [ ] ship electron app
- ipfs/distribution html pages
  - [ ] generate pages
  - [ ] design pages
  - [ ] gather assets (screenshots)
  - [ ] ship ipfs/distribution
  - [ ] ipfs/distribution blog post
- [ ] get go-ipld ready for go-ipfs
- [ ] multicodec + go-multicodec
- add improvements
  - [ ] improve go-ipfs add to use dagutils.Editor
  - [ ] go-ipfs: commands/files TarFile
  - [ ] node-ipfs-api: use tar-stream and tar-fs

### @diasdavid

- electron-app
  - [~] Add a settings menu https://github.com/ipfs/electron-app/issues/28
  - [x] Upload several files at once (waiting on multipart/tar stream) - https://github.com/ipfs/electron-app/issues/26
  - [x] Add UI element to close the app - https://github.com/ipfs/electron-app/issues/25
  - [x] Fix peer count - https://github.com/ipfs/electron-app/issues/22
  - [~] Keyboard shortcuts - https://github.com/ipfs/electron-app/issues/31
- node-ipfs-api
  - [x] restructure/clean codebase https://github.com/ipfs/node-ipfs-api/pull/44

### @mappum

- 

### @krl

- [/] starlog
- [/] feed webcomponent
- [ ] following/polling of other feeds

### @lgierth

- TLS on gateway.ipfs.io https://github.com/ipfs/infrastructure/issues/6
  - [x] store secrets encrypted in github https://github.com/ipfs/infrastructure/pull/60
  - [x] setup https://github.com/ipfs/infrastructure/pull/61
  - [ ] properly signed certificate
- Static pages on ipfs.io
  - [x] fix IPNSHostname in gateway https://github.com/ipfs/go-ipfs/pull/1577
  - [ ] Migrate gateway to ipfs.io https://github.com/ipfs/infrastructure/issues/4
  - [ ] dist.ipfs.io and ipfs.io/dist https://github.com/ipfs/distributions
- [ ] one-click deploys to digitalocean https://github.com/seven1m/do-install-button
- storage
  - [ ] big hetzner node https://github.com/ipfs/infrastructure/pull/62
  - [ ] storagebot

**Comment by jbenet** [comment]
Etherpad for this week: https://etherpad.mozilla.org/OXahDHFZOU

**Comment by krl** [comment]
had a terrible week of hardware failure, that i still have not been able to get around (have to use my screenless old x61 with external monitor atm) have however made progress on webcomponents and starlog concept.

The webcomponent feed-view can now display an infinite-scroll list of a feed. ( here /ipfs/QmXUzXsqFgKfS5MBXzn2pfaAbxWQzirKSvFM45ywEMWeD8/app/ )

What's left is appending, publishing to ipns, and muxing multiple feeds together for a timeline view. and then we have a webcomponent-powered twitter-clone on ipfs.

I think i'll still have time to get something installable until camp, i've gotten over the most hurdles. and my new computer should arrive tomorrow.

**Comment by lgierth** [comment]
@krl could you move that sprint update to #25 instead?

**Comment by whyrusleeping** [comment]
@diasdavid this is the wrong issue, Sprint S is the current one.

**Comment by diasdavid** [comment]
@whyrusleeping we were missing a issue
![image](https://cloud.githubusercontent.com/assets/1211152/9208274/e98b78b4-4072-11e5-8580-ce961a8a4776.png)

Sprint S was August 3
Sprint T is August 10
Sprint U is August 17

**Comment by jbenet** [comment]
### Infrastructure discussion
- tls on gateway
- static pages on gateway
  - migrate to ipfs.io
  - dist.ipfs.io and ipfs.io/dist
- one-click deploys to DO
- storage node (hezner?)
  - storagebot 

Stuff to buy
- computer (nuc) gigabit ethernet, usb 3.0
- US to Europe power adapters (10)
- lots of duct + paper tape
- ethernet cables (3 x 5m)
- extension cords (5)
- power strips (5)

**Comment by jbenet** [comment]
@whyrusleeping @diasdavid this is the current issue. renamed **"Sprint 8/10"**

**Comment by diasdavid** [comment]
Cool :) Let's go with month `name + day`, because seeing 8/10 instead of 10/8 for me (and probably everyone from outside of the US) is very confusing :)

**Comment by jbenet** [comment]
@diasdavid ok sounds good! (i guess i shouldn't suggest a unix timestamp range then...)

**Comment by diasdavid** [comment]
# SPRINT UPDATE

CCC was awesome, had the chance to refresh my knowledge in RF stuff and learn a lot of new things. Excited about the Rad1o Badge, HackRF and what we can do with it (Radio School™).

The focus for the last sprint was mainly getting stuff ready for people in CCC to use:

electron-app
  - [~] Add a settings menu https://github.com/ipfs/electron-app/issues/28
  - [x] Upload several files at once (waiting on multipart/tar stream) - https://github.com/ipfs/electron-app/issues/26
  - [x] Add UI element to close the app - https://github.com/ipfs/electron-app/issues/25
  - [x] Fix peer count - https://github.com/ipfs/electron-app/issues/22
  - [~] Keyboard shortcuts - https://github.com/ipfs/electron-app/issues/31
- node-ipfs-api
  - [x] restructure/clean codebase https://github.com/ipfs/node-ipfs-api/pull/44

I've organized two milestones for the electron-app, the 0.1.0 being all the features above + UI polish and the 0.2.0 being a new release, right on time for Container Camp, with more features but also with bugs fixed we might find in between. 

Other random stuff includes trying to get the NUCs going, webui patch and more little things here and there.

**Comment by lgierth** [comment]
- TLS on gateway.ipfs.io
  - [x] store secrets encrypted in github
  - [x] https://gateway.ipfs.io
  - [ ] properly signed certificate
- Static pages on ipfs.io
  - [x] fix IPNSHostname in gateway (ready for merge)
  - [ ] Migrate gateway to ipfs.io
  - [ ] dist.ipfs.io and ipfs.io/dist
  - static pages are very close, but need more work in ipns to be really useful (how to do updates, etc.)
- storage
  - [x] got 3 TB node at hetzner, called pollux
  - [ ] mirroring downloads.openwrt.org, about to run `ipfs add -s rabin`
  - [ ] storagebot
- cccamp
  - got equipment, and two NUC computers, which we didn't get set up in the end
  - the NUCs can live on as cjdns+ipfs nodes at freifunk backbone sites
  - [ ] one-click deploys to digitalocean, didn't get done

**Comment by jbenet** [comment]
### @jbenet's SPRINT UPDATE

- [x] CCC -- a ton of things here. (travel is hugely time consuming.)
  - [x] get lots of great user feedback
  - [ ] workshop (interrupted by thunder)
  - [x] guided many people through installs. (collected concerns for dist.ipfs.io)
- infrastructure (FAIL)
  - [ ] ssl cert for gateway.ipfs.io -- Not yet done. never got email from rapidssl. will email again.
- electron app (FAIL)
  - [ ] design dropdown menu -- FAIL
  - [ ] implement dropdown menu -- FAIL
  - [ ] ship electron app -- FAIL
- ipfs/distribution html pages (this took longer than expected) (ALMOST SUCCESS (FAIL))
  - [x] generate pages -- done!
  - [x] design pages -- done!
  - [x] gather assets (screenshots) -- done! 
  - [~] ship ipfs/distribution -- waiting on electron-app dist
  - [ ] ipfs/distribution blog post -- nope.
- [x] multicodec + go-multicodec -- done, minus test errors (SUCCESS)
- [~] get go-ipld ready for go-ipfs -- almost! (ALMOST SUCCESS (FAIL))
- add improvements (SUCCESS (as much as we needed))
  - [x] improve go-ipfs add to use dagutils.Editor -- done
  - [~] go-ipfs: commands/files TarFile -- began but not done
  - [ ] node-ipfs-api: use tar-stream and tar-fs -- not done. but no longer needed now.
- [x] fix a bunch of vinyl problems -- this took forever -- (SUCCESS)
- [x] shipped https://github.com/jbenet/asciinema-selfhost
- [x] simple federated wiki integration -- (SUCCESS) 
- [x] downloaded all of openwrt
- [x] got sirius (node to match pollux)
- [x] made pinbot able to change friends live

**Comment by whyrusleeping** [comment]
This week consisted of a bunch of codebase cleanup type changes. I refactored the node construction routine, and made the blockservice constructor not return an error (programmer errors should panic, not error). 
the rabin chunking PR merged early in the sprint, and it works pretty well. To get better compression out of it, some tweaking of parameters may be necessary.
I fixed the dht commands so they now show more information about the queries. This will be helpful when we get to refactoring the dht and writing ipns changes.
Also fixed a slight bug in the ipfs refs command where the json objects contained newlines in strings.
The dag editor is a little more efficient now, intermediate nodes wont be added to the primary dagstore (and therefore, not provided).

**Comment by jbenet** [comment]
@rht @krl @kbala (and anyone else) drop update here pls, thanks!

**Comment by rht** [comment]
On the refactor train: ipfs get and context.Context's.
Add readonly (the safe ones) API from commands to gateway (discovered that gateway.ipfs.io has 137k hashes).
GC PR still WIP, though the refactor part of this can be pulled away into a smaller PR.
Made a runtime memory leak test for https://github.com/ipfs/go-ipfs/issues/1584.

**Comment by heems** [comment]
This week I worked on outlining decision making in bitswap https://github.com/ipfs/go-ipfs/pull/1590.
I also refactored some stuff in bssim and added strategy graphs so you can compare the performance of different strategies (though it's not very useful atm since I haven't finished that pull request).
And I started working on a better way to create new graph types in bssim, but I'm not sure if it will work out yet.
Also tweaked the bitswap_wo_routing test to actually not use routing.

**Comment by rht** [comment]
@heems what is a strategy graph used for? https://github.com/heems/bssim/commit/6b3f8269c4332e4b14d9b09307f699370bdc1263.

Is it the decision engine itself, or the graph out of a decision engine on a specific system?

**Comment by heems** [comment]
It's the graph out of a decision engine on a specific system if I understand that correctly.  So you can run `./bssim -str Mean` and `./bssim -str Nice` (where Mean and Nice are peer_request_queue sorting strategies) a bunch of times and it will compare how long they took to complete.  It compares different decision engines.

**Comment by krl** [comment]
Update:

Sorry for missing the sprint, got home exausted and slept through it, here's my update though: 

Have been working on starlog and webcomponents. Created a ipfs-add-file webcomponent using drag-drop. UI improvements and lots of behind-the-scene datastructure work.

Will finish a ui for the starlog project this week. Then i want to begin working on pulling the webcomponents out into its own repo, and incorperating them into the webui.
